### PR TITLE
added hardwareBackEnabled option

### PR DESCRIPTION
**Pull request by thebigredgeek**
This change allows app code to disable to hardware back button hook.  This is useful if you want to programmatically disable the default back button behavior on certain views but not others.

### CODE REVIEW

File: src/navigators/DrawerNavigator.js [hunk 1]
```diff
@@ -84,7 +84,7 @@ const DrawerNavigator = (
     />
   ));
 
-  return createNavigationContainer(navigator, containerConfig);
+  return createNavigationContainer(navigator, mergedConfig);
```
Comment (skevy) [inline src/navigators/DrawerNavigator.js]: Wow @thebigredgeek has this just been broken this whole time?

**Comment by skevy** [comment]
Looks like you introduced a few Flow errors here, @thebigredgeek

File: src/navigators/DrawerNavigator.js (hunk 1 above)
Comment (thebigredgeek) [inline src/navigators/DrawerNavigator.js]: I believe so :(

**Comment by thebigredgeek** [comment]
@skevy I'll see if I can fix it this week or next

File: src/createNavigationContainer.js [hunk 2]
```diff
@@ -153,9 +155,11 @@ export default function createNavigationContainer<T: *>(
         return;
       }
 
-      this.subs = BackAndroid.addEventListener('backPress', () =>
-        this.dispatch(NavigationActions.back())
-      );
+      if (hardwareBackHandled !== false) {
```
Comment (andry-baka) [inline src/createNavigationContainer.js]: For props name, how about `backAndroidHandled`to make it consistent with next line `BackAndroid.addEventListener`?

**Comment by anton6** [comment]
Hi, is there any progress on this PR? thanks

**Comment by thebigredgeek** [comment]
I need to find time for this.  i started a new job and am no longer working with React Native or this lib at all.  If someone wants to take this over, they are more than welcome

**Comment by brentvatne** [comment]
moved discussion over to https://github.com/react-navigation/rfcs/issues/3

**Pull request closed without merging**